src/main/java/com/seattleacademy/team20/SkillController.java

```diff
@@ -1,21 +1,191 @@
+    uploadSkill(skills);
+
+    return "skill-upload";
+  }
+
+  public List<SkillInfo> selectSkills() {
+    final String sql = "select * from skills";
+    return jdbcTemplate.query(sql, new RowMapper<SkillInfo>() {
+      public SkillInfo mapRow(ResultSet rs, int rowNum) throws SQLException {
+        return new SkillInfo(rs.getString("category"),
+            rs.getString("name"), rs.getInt("score"));
+      }
+    });
+  }
+
+  private FirebaseApp app;
+
+  public void initialize() throws IOException {
+    FileInputStream refreshToken = new FileInputStream(
+        "/Users/hatanyuto/test/dev-yuto-hatano-firebase-adminsdk-bgvtt-416796d4de.json");
+
+    FirebaseOptions options = new FirebaseOptions.Builder()
+        .setCredentials(GoogleCredentials.fromStream(refreshToken))
+        .setDatabaseUrl("https://dev-yuto-hatano.firebaseio.com/")
+        .build();
+    app = FirebaseApp.initializeApp(options, "other");
+  }
+
+  public void uploadSkill(List<SkillInfo> skillList) {
+    final FirebaseDatabase database = FirebaseDatabase.getInstance(app);
+    DatabaseReference ref = database.getReference("/skillcategory");
+
+    //    複合キーの作成
+    //    Function<Skills, String> compositeKey = prd -> {
+    //      StringBuffer sb = new StringBuffer();
+    //      sb.append(prd.getName()).append("-").append(prd.getScore());
+    //      return sb.toString();
+    //    };
+
+    //		データの取得
+    //    List<Map<String, Object>> datalist = new ArrayList<Map<String, Object>>();
+    //    Map<String, Object> map;
+    //    Map<String, Map<String, List<Skills>>> skillMap = skillList.stream().collect(
+    //        Collectors.groupingBy(Skills::getCategory,
+    //            Collectors.groupingBy(compositeKey)));
+    //    for (Entry<String, Map<String, List<Skills>>> entry : skillMap.entrySet()) {
+
+    //    List<Map<String, Object>> datalist = new ArrayList<Map<String, Object>>();
+    Map<String, Object> map = new HashMap<>();
+    Map<String, List<SkillInfo>> skillMap = skillList.stream().collect(Collectors.groupingBy(SkillInfo::getCategory));
+    //    Arraylist使うとき、要素数わかってる場合は指定する
+    List<Map<String, Object>> list = new ArrayList<>(map.size());
+    Map<String, Object> innerMap;
+
+    String[] categories = { "front_end", "back_end", "dev_Ops" };
+    for (String category : categories) {
+      innerMap = new HashMap<>();
+      innerMap.put("category", category);
+      innerMap.put("skills", skillMap.get(category).stream().map(s -> s.getSkills()).collect(Collectors.toList()));
+      list.add(innerMap);
+      //    for (Entry<String, List<SkillInfo>> entry : skillMap.entrySet()) {
+      //      map = new HashMap<>();
+      //      map.put("category", entry.getKey());
+      //      map.put("skills", entry.getValue().stream().map(s -> s.getSkills()).collect(Collectors.toList()));
+      //
+      //      datalist.add(map);
+      //      switch文にbreakは必須！！
+      //      switch ("entry.getKey()") {
```

Comment:
```suggestion
      //      switch (entry.getKey()) {
```
ダブルクォーテつけると文字列になってしまうよ

Reply:
コメントアウトしているので、対応不要ですが確認しておいてくださいー